perceval/utils/density_matrix.py

```diff
@@ -230,7 +268,9 @@ def _bra_str(bs: BasicState):
 
     def to_svd(self, threshold: float = 1e-8, batch_size: int = 1):
```

Comment:
Since this method is a bit complex (and long ^^'), you could refactor into two method called by this one:
`_to_svd_with_array_eigh`
and 
`_to_svd_with_sparse_eigsh`